src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -2984,6 +2984,28 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                 for( size_t i = 0; i < const_tuple->n_elements; i++ ) {
                     // TODO: Adjust for global variables as well
                     // See if-else check on llvm_symtab for the logic
+                    if (ASR::is_a<ASR::ListItem_t>(*const_tuple->m_elements[i])) {
+                        ASR::ListItem_t* list_ele = ASR::down_cast<ASR::ListItem_t>(const_tuple->m_elements[i]);
+                        uint64_t ptr_loads_copy = ptr_loads;
+                        ptr_loads = 0;
+                        ASR::Variable_t *target_var = EXPR2VAR(list_ele->m_a);
+                        uint32_t h = get_hash((ASR::asr_t*)target_var);
+                        llvm::Value* target_ptr = nullptr;
+                        if (llvm_symtab.find(h) != llvm_symtab.end()) {
+                            target_ptr = llvm_symtab[h];
+                            if (ASR::is_a<ASR::Pointer_t>(*target_var->m_type)) {
+                                target_ptr = CreateLoad(target_ptr);
+                            }
+                        } else {
+                            throw CodeGenError("List Variable is not defined");
```

Comment:
I think instead of directly querying the `llvm_symtab`, we should first try out `visit_expr(*const_tuple->m_elements[i])`.

Reply:
The `ListVariable` will be already defined, so I thought to save some time by avoiding it to visit repeatedly.

Reply:
I think `Assignment(TupleConstant, TupleConstant)` should be handled separately. It has very different semantics from `Assignment(Tuple, Tuple)`. In `Assignment(TupleConstant, TupleConstant)` first all the values should be deepcopied and stored into a vector. Then, these should be written. For example,

```
a, b = b, a
```

If you do `a = b` first then `b = a` won't work because `a` has taken the value of `b`.  So `a` and `b` will both attain the same value which is not something we want. Instead we should do `d1 = b, d2 = a` then, `a = d1, b = d2`. Note that a `d1, d2` denote deepcopies of `a` and `b`.

Reply:
> The `ListVariable` will be already defined, so I thought to save some time by avoiding it to visit repeatedly.

Its not necessarily a variable. Imagine, `tuplev[1][j], tuplev[1][j + 1] = tuplev[1][j + 1], tuplev[1][j]`. I mean it can be anything. So that's why visiting the expression is a better idea. You can enforce a pointer by setting `ptr_loads = 0` or `ptr_loads = 1`. If you want a value then `ptr_loads = 2`.